grift/grift.cabal

```diff
@@ -31,14 +34,15 @@ library
                      , GRIFT.InstructionSet.Priv
   build-depends:       array >= 0.5 && < 0.6
                      , base >= 4.7 && < 5
-                     , bv-sized >= 0.7 && < 0.8
-                     , bv-sized-float
-                     , bytestring >= 0.10.8 && < 0.10.9
+                     , bv-sized >= 1.0.2 && < 1.0.3
+                     , bv-sized-float >= 0.2 && < 0.3
+                     , bytestring >= 0.10.8 && < 0.10.13
                      , containers >= 0.5.11 && < 0.7
                      -- , crucible
                      -- , crucible-llvm
                      , elf-edit
                      , filepath >= 1.4 && < 1.5
+                     , ghc-typelits-natnormalise
```

Comment:
I'm a tiny bit worried about a dependency like this - what came up that requires it? I'm not opposed if we need it, but I just worry about how well that sort of thing will be maintained over time

Reply:
Yeah I don't love it.

I looked into using this because otherwise constraints like `64 <= w` do not imply constraints like `32 <= w`, and so most of the semantics would need constraints like `(5 <= w, 12 <= w, 32 <= w, 64 <= w)`.

Alternatively we might be able to do something like `widen :: (smaller <= larger, larger <= w) => ((smaller <= w) => a) -> a` and wrap at problematic call sites? :-\